src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -252,3 +255,137 @@ def test_WINNF_FT_S_SCS_15(self,config_filename):
 
     # Check registration response
     self.assertEqual(response['response']['responseCode'],104)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_17(self):
+    """Invalid certificate following an approved registration request
+       Checks that SAS UUT response with fatal alert message.
+    """
+    device_cert_name = "short_lived_client"
+    expiry_timer = 2  # in minutes
+
+    self.create_short_lived_certificate("CBSD", device_cert_name, expiry_timer)
+
+    device_cert = self.getCertFilename(device_cert_name + ".cert")
+    device_key = self.getCertFilename(device_cert_name + ".key")
+    wait_timer = (expiry_timer * 60) + 10  # In seconds
+
+    # Successful Tls Handshake
+    self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'],
+                                   device_cert, device_key)
+    logging.info("TLS Handshake Succeeded")
+
+    # Load the device_a file
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+
+    with security_testcase.CiphersOverload(self._sas, ['AES128-GCM-SHA256'],
+                                           device_cert,
+                                           device_key):
+      self.assertRegistered([device_a])
+
+    logging.info("CBSD device is in registered state")
+    logging.info("Waiting for %s secs so the certificate will become invalid" % wait_timer)
+
+    # sleep for waitTimer so that certificate will become invalid
+    time.sleep(wait_timer)
```

Comment:
People will be disincentivized from running a test that sleeps for 2+ minutes.  Can this be done reliably in less time?

Here's an idea for a better algorithm:
- Assert that the first request succeeds.
- Retry the request in a loop, with a 1-second `sleep`.
- Assert that the number of iterations before failure is within reasonable bounds.

That will allow the test to finish ASAP, and immediate feedback will make it easier to tune the certificate lifetime.

Reply:
The   time duration has been reduced. Certificate expiry time has been changed from   2 minutes to 1 minute.          As we know the time at which the certificate will expire, I think we do not   have to run requests in a loop until it fails. Also the test case description   does not expect multiple TLS handshakes to happen. The current algorithm just   waits until the time the certificate expires (a 5 sec buffer is also added)   and then tries the TLS handshake again and verifies that the TLS handshake   fails.